src/frontend/src/binder/relation/table_or_source.rs

```diff
@@ -70,14 +70,32 @@ impl BoundSource {
 }
 
 impl Binder {
+    pub fn bind_catalog_relation_by_object_name(
+        &mut self,
+        object_name: ObjectName,
+        bind_creating_relations: bool,
+    ) -> Result<Relation> {
+        let (schema_name, table_name) =
+            Binder::resolve_schema_qualified_name(&self.db_name, object_name)?;
+        self.bind_catalog_relation_by_name(
+            None,
+            schema_name.as_deref(),
+            &table_name,
+            None,
+            None,
+            bind_creating_relations,
+        )
+    }
+
     /// Binds table or source, or logical view according to what we get from the catalog.
-    pub fn bind_relation_by_name_inner(
+    pub fn bind_catalog_relation_by_name(
```

Comment:
Why to rename this? (And it becomes inconsistent with others like bind_table)

Reply:
The original name `bind_relation_by_name_inner` doesn't indicate what this function does. We have to look into the implementation details, to understand that it actually just binds a catalog relation by name.

A catalog relation can be one of:
- A table catalog (corresponding to mv, table, index)
- A source catalog
- A view catalog
- A system table catalog

> And it becomes inconsistent with others like bind_table

Could you elaborate? Perhaps I can find a better name for it.

Reply:
But why "relation" is not the same as "catalog relation"

Reply:
`bind_relation_by_name` calls `bind_relation_by_name_inner`.

Inside `bind_relation_by_name`, it splits into two branches. The first branch binds a `cte` relation, i.e. a relation that's instantiated within the query itself, rather than corresponding to some existing relation in from our catalogs.

In the other branch, we bind a catalog relation or a "base relation", one that existed prior the the query being bound.

For describe fragments, we only care about the latter branch, catalog relation.